core/services/workflows/syncer/handler.go

```diff
@@ -586,6 +585,226 @@ func (h *eventHandler) tryEngineCleanup(wfID string) error {
 	return nil
 }
 
+// registerWorkflow handles creating a workflow from contract workflow metadata. This method must remain idempotent.
+func (h *eventHandler) registerWorkflow(
+	ctx context.Context,
+	wfMeta GetWorkflowMetadata,
+) error {
+	wfID := hex.EncodeToString(wfMeta.WorkflowID[:])
+	owner := hex.EncodeToString(wfMeta.Owner)
+
+	// If there is not a workflow spec in the database, then fetch artifacts and create it
+	spec, err := h.workflowArtifactsStore.GetWorkflowSpec(ctx, owner, wfMeta.WorkflowName)
+	if err != nil {
+		decodedBinary, config, err := h.workflowArtifactsStore.FetchWorkflowArtifacts(ctx, wfID, wfMeta.BinaryURL, wfMeta.ConfigURL)
```

Comment:
Consider moving all of this under a function called "createWorkflowSpec()" or similar

Reply:
Agreed

Reply:
To be changed in subsequent PR that touches handlers.